assignment.rb

```diff
@@ -0,0 +1,310 @@
+                  print "How much would you like to withdraw? $"
+                  amount = gets.chomp.to_i
+                  puts ""
+                  # Let's make sure the account and the ATM can handle this transaction
+                  if user.can_withdraw?(amount) && @atm.can_withdraw?(amount)
+                    puts "Dispensing cash..." 
+                    user.deduct(amount)
+                    @atm.deduct(amount)
+                    puts "Your new balance is $#{user.balance}"
+                    puts "Thank you for paying that enourmous fee."
+                    puts ""
+                  elsif user.can_withdraw?(amount) == false
+                    puts "You have asked to withdraw more than you have in your account"
+                    puts ""
+                  elsif @atm.can_withdraw?(amount) == false
+                    puts "ATM has insufficient balance to fulfill this request."
+                    puts ""
+                  else
+                    puts "Something went wrong, and I don't know what"
+                    puts ""
+                  end
+                # The user would like to exit their account.  
+                else
+                  puts "Now exiting to the main menu."
+                  puts ""
+                  display_acc_menu = false
+                  main_men_sel_valid = false
+                end
+              # Invalid input in the account menu. Log out for security
+              else
+                puts "This was not a valid menu input."
+                puts "Logging you out for security purposes."
+                puts ""
+                display_acc_menu = false
+                main_men_sel_valid = false
+              end
+            end
+          # Login did not match records. 
+          else
+            puts "Your login does not match our records."
+            puts "Please try loggin in again."
+            puts ""
+            main_men_sel_valid = false
+          end
+          
+        when 2
+          # This code should view the atm's current balance
+          puts "This ATM currently has $#{@atm.balance}"
+          puts ""
+          main_men_sel_valid = false
+        
+        else
+          # The user would like to exit the program entirely.
+          puts "Now exiting the ATM program. Goobye."
+          puts ""
+          main_men_sel_valid = false
+          self.exit_prog = true
+        end
+      end
+    # Invalid input in the main menu. Request another attempt.  
+    else
+      puts "Your input was invalid. Please try again."
+      puts ""
+    end 
+  end
+
+  # Displays the main menu and returns the user's selection
+  def main_menu
+    puts "What would you like to do?"
+    puts ""
+    puts "  1: Enter your account information"
+    puts "  2: View the ATM's cash balance"
+    puts "  3: Shut down this ATM"
+    puts ""
+    print "Please enter your selection: "
+    main_menu_sel = gets.chomp.to_i
+    puts ""
+    return main_menu_sel
```

Comment:
as we talked about yesterday, the ruby style doesn't call `return` unless it's trying to exit a method prematurely